document/src/main/java/com/ritense/document/domain/impl/JsonSchemaDocument.java

```diff
@@ -197,7 +197,7 @@ public static CreateDocumentResultImpl create(
      * @param versionCheck    The version on which the content was based on (in other words: this's ver)
      * @return Object representing the result of the operation (either resulting document or errors)
      */
-    public synchronized ModifyDocumentResultImpl applyModifiedContent(
+    public ModifyDocumentResultImpl applyModifiedContent(
```

Comment:
@theo-ritense 
> This fix opens up the possibility that 2 parallel job/threads can change the document, which might lead to data loss.
No it doesn't. It will then throw an error in one of the 2 parallel jobs: [ConflictedDocumentVersion](https://github.com/valtimo-platform/valtimo-backend-libraries/blob/development/11.2.0/document/src/main/java/com/ritense/document/domain/impl/JsonSchemaDocument.java#L210)

Reply:
This actually proves my point. I can still call the `applyModifiedContent` in parallel on the same version. This check only works properly after the transaction finishes, which, with some incorrectly modeled processes, can take quite long. 

Before the transaction finishes, one can still retrieve the same version and perform an update on it.

I know: it's hypothetical, but certainly not impossible.